misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/utils/PackageUtils.java

```diff
@@ -329,7 +329,7 @@ private static String getFullModuleName(Document document) {
         }
 
         String moduleNamePart = document.module().moduleName().moduleNamePart();
-        return moduleNamePart != null ? packageNamePart + "." + encodeModuleName(moduleNamePart) : packageNamePart;
+        return moduleNamePart != null ? encodeModuleName(packageNamePart + "." + moduleNamePart) : packageNamePart;
```

Comment:
Why do we encode the package name as well?

Reply:
Because when generating java classes during runtime code generation, the java class name will be as `<packageName.<module Name>.<file name>` and therefore the entire string gets encoded.